src/server/components/charts-engine/components/processor/isolated-sandbox/interop/libs/ql-chart-v1.ts

```diff
@@ -28,28 +30,36 @@ export const libsQlChartV1Interop: LibsInterop = {
         jail.setSync(`${NAMESPACE}_buildGraph`, ((arg: string) => {
             const parsedArg = JSON.parse(arg) as Parameters<typeof qlModule.buildGraph>[0];
             parsedArg.ChartEditor = chartEditorApi;
-            const result = qlModule.buildGraph(parsedArg);
+            const result = functionTimeout(qlModule.buildGraph, {
+                timeout: JS_EXECUTION_TIMEOUT,
+            })(parsedArg);
             return JSON.stringify(result);
         }) satisfies LibsQlChartV1BuildGraph);
 
         jail.setSync(`${NAMESPACE}_buildLibraryConfig`, ((arg: string) => {
             const parsedArg = JSON.parse(arg) as Parameters<typeof qlModule.buildLibraryConfig>[0];
             parsedArg.ChartEditor = chartEditorApi;
-            const result = qlModule.buildLibraryConfig(parsedArg);
+            const result = functionTimeout(qlModule.buildLibraryConfig, {
+                timeout: JS_EXECUTION_TIMEOUT,
+            })(parsedArg);
             return JSON.stringify(result);
         }) satisfies LibsQlChartV1BuildLibraryConfig);
 
         jail.setSync(`${NAMESPACE}_buildChartsConfig`, ((arg: string) => {
             const parsedArg = JSON.parse(arg) as Parameters<typeof qlModule.buildChartsConfig>[0];
             parsedArg.ChartEditor = chartEditorApi;
-            const result = qlModule.buildChartsConfig(parsedArg);
+            const result = functionTimeout(qlModule.buildChartsConfig, {
+                timeout: JS_EXECUTION_TIMEOUT,
```

Comment:
You only need JS_EXECUTION_TIMEOUT to execute the `js` tab
Maybe you should use ONE_SECOND for everything else?

Reply:
yep, i refactored it to timeout param